controllers/main.go

```diff
@@ -2119,6 +2160,220 @@ func (controller *MainController) IsValidVoteBits(userVoteBits uint16) bool {
+	err = feeTx.FromBytes(feeTxBytes)
+	if err != nil {
+		return nil, errors.New("unable to deserialize transaction")
+	}
+
+	dbMap := controller.GetDbMap(c)
+	validFeeAddrs, err := models.GetInactiveFeeAddresses(dbMap)
+	if err != nil {
+		log.Infof("database error: %v", err)
+		return nil, errors.New("database error")
+	}
+
+	var feeAddr string
+	var feeAmount dcrutil.Amount
+	const scriptVersion = 0
+findAddress:
+	for _, txOut := range feeTx.TxOut {
+		_, addresses, _, err := txscript.ExtractPkScriptAddrs(scriptVersion,
+			txOut.PkScript, controller.Cfg.NetParams)
+		if err != nil {
+			log.Infof("Extract: %v", err)
+			return nil, err
+		}
+		for _, addr := range addresses {
+			addrStr := addr.Address()
+			for _, validFeeAddr := range validFeeAddrs {
+				if addrStr == validFeeAddr {
+					feeAddr = validFeeAddr
+					feeAmount = dcrutil.Amount(txOut.Value)
+					break findAddress
+				}
+			}
+		}
+	}
+	if feeAddr == "" {
+		log.Infof("feeTx did not invalid any payments")
+		return nil, errors.New("feeTx did not include any payments")
+	}
+
+	feeEntry, err := models.GetFeesByFeeAddress(dbMap, feeAddr)
+	if err != nil {
+		log.Infof("GetFeeByAddress: %v", err)
+		return nil, errors.New("database error")
+	}
+	voteAddr, err := dcrutil.DecodeAddress(feeEntry.Address, controller.Cfg.NetParams)
+	if err != nil {
+		log.Errorf("PayFee: DecodeAddress: %v", err)
+		return nil, errors.New("database error")
+	}
+	_, err = dcrutil.NewAddressPubKeyHash(dcrutil.Hash160(votingWIF.PubKey()), controller.Cfg.NetParams,
+		dcrec.STEcdsaSecp256k1)
+	if err != nil {
+		log.Errorf("PayFee: NewAddressPubKeyHash: %v", err)
+		return nil, errors.New("failed to deserialize voting wif")
+	}
+	// TODO: validate votingkey against ticket submission address
+
+	sDiff := dcrutil.Amount(feeEntry.SDiff)
+	// TODO - wallet relayfee
+	relayFee, err := dcrutil.NewAmount(0.0001)
+	if err != nil {
+		log.Errorf("PayFee: failed to NewAmount: %v", err)
+		return nil, errors.New("internal error")
+	}
+
+	minFee := txrules.StakePoolTicketFee(sDiff, relayFee, int32(feeEntry.BlockHeight), controller.Cfg.PoolFees, controller.Cfg.NetParams)
+	if feeAmount < minFee {
+		log.Infof("too cheap: %v %v", feeAmount, minFee)
+		return nil, fmt.Errorf("dont get cheap on me, dodgson (sent:%v required:%v)", feeAmount, minFee)
+	}
+
+	// Get vote tx to give to wallet
+	ticketHash, err := chainhash.NewHashFromStr(feeEntry.TicketHash)
+	if err != nil {
+		log.Errorf("PayFee: NewHashFromStr: %v", err)
+		return nil, errors.New("internal error")
+	}
+
+	now := time.Now()
+	resp, err := controller.Cfg.StakepooldServers.PayFee(r.Context(), ticketHash, votingWIF, dcrutil.NewTx(feeTx))
+	if err != nil {
+		log.Errorf("PayFee: %v", err)
+		return nil, errors.New("RPC server error")
+	}
+
+	err = models.InsertFeeAddressVotingKey(dbMap, voteAddr.Address(), votingWIF.String(), voteBits)
+	if err != nil {
+		log.Errorf("PayFee: InsertVotingKey failed: %v", err)
+		return nil, errors.New("internal error")
```

Comment:
I've seen return `internal error`, `internal database error` and `database error`. Could harmonize all into a reusable `var dbErr = errors.New("internal database error")` or a more generic `var internalServerErr = errors.New("internal error")`.